### ISSUE TITLE
No such file or directory: 'data/id.zh.train.llama'

### ISSUE THREAD

**Issue by Z1zs** [RUCAIBox/Language-Specific-Neurons#2]
Hi, thanks for your amazing work!
I tried to run activation.py as described in readme.md, but get the error:
_No such file or directory: 'data/id.zh.train.llama'_

I'm wondering what's the structure of **id.zh.train.llama** and how should I organize it. Or will you please publish it in the future?

**Comment by StevenTang1998** [comment]
The type of this checkpoint is `List[int]`. We tokenize the wiki texts and concat them to get this file. I believe it is easy to reproduce.
Since each file is over 1GB, we may not publish them.

**Comment by Z1zs** [comment]
Thank you for the reply, I have reproduced successfully.

**Comment by Z1zs** [comment]
Hi , I got another problem.
The length of tokenized sequences in wikipedia dataset varies, how can you concat tensor of different size? 
If padding and truncation is used during producing the id-file, which max_length should I set when I tokenize the dataset?(currently I set it by 4096, but not sure if it's proper)
The lines for tokenizing look like this:

```
>     ids=[]
>     for i in tqdm(range(rows)):
>         tokens_prompt = tokenizer(data[i]['text'], return_tensors="pt",padding='max_length',max_length=4096,truncation=True).input_ids[0]
>         ids.append(tokens_prompt)
>     ids=torch.stack(ids)
```

is there anything wrong with it?
Thank you very much again in advance.

**Comment by StevenTang1998** [comment]
Just concat all the ids with any padding then reshape it to size of `(-1, max_len)`.
This is the packing strategy used in pre-training. You can refer to RoBERTa or T5 for details.

**Comment by Z1zs** [comment]
I get it! Thanks a lot for your reply!

**Comment by Z1zs** [comment]
I get it! Thanks a lot for your reply!

**Comment by lizhou21** [comment]
> The type of this checkpoint is `List[int]`. We tokenize the wiki texts and concat them to get this file. I believe it is easy to reproduce. Since each file is over 1GB, we may not publish them.

Do you mind sharing this code directly? Thanks!
